icu4c/source/i18n/unicode/numberrangeformatter.h

```diff
@@ -612,8 +613,9 @@ class U_I18N_API LocalizedNumberRangeFormatter
     ~LocalizedNumberRangeFormatter();
 
   private:
-    // TODO: This is not thread-safe! Do NOT check this in without an atomic here.
-    impl::NumberRangeFormatterImpl* fImpl = nullptr;
+    std::atomic<impl::NumberRangeFormatterImpl*> fAtomicFormatter = {};
```

Comment:
okay, I don't see `fAtomicFormatter` called from an inline environment. So I don't see a much of an argument for including this in a public header.  Can the atomicity be implemented entirely within the .cpp side? Perhaps with an opaque type here in the class definition?

Reply:
Yes, it can.  But how to you make an opaque type?  We don't know how big the std::atomic needs to be.